ui/pages/requests/component/address_bar.go

```diff
@@ -88,7 +89,15 @@ func (a *AddressBar) Layout(gtx layout.Context, theme *chapartheme.Theme) layout
 		if !ok {
 			break
 		}
-		if _, ok := event.(widget.ChangeEvent); ok {
+
+		switch event.(type) {
+		// on carriage return event
+		case widget.SubmitEvent:
+			if a.onSubmit != nil {
+				go a.onSubmit()
```

Comment:
Just a remark, I could be totally wrong. Hut I don't understand why this on use a subroutine while the on change doesn't.

Maybe it's normal and a comment would be a good idea

Reply:
Thanks for the swift comment. In all honesty I copied the implementation in lines 137/138. Since this function runs when the UI is updated, it boils down to the onSubmit being a costly function. 

For example, imagine clicking submit on an endpoint which times out after 10s. If this request (onSubmit) was not concurrent then none of the UI would be able to update until the request is finished.

Reply:
a comment may help then, maybe

Reply:
Done, thanks for the feedback